File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer>fre =new HashMap<>();
        if(s.length()!=t.length()){
            return false;
        }
        for(char c:s.toCharArray()){
            fre.put(c,fre.getOrDefault(c,0)+1);
        }
        for(char c:t.toCharArray()){           
            if(fre.containsKey(c) && fre.get(c)>0){ 
                fre.put(c,fre.get(c)-1);
            }
        }
        for(char n:fre.keySet()){
             if(fre.get(n)>0){
                return false;
             }
        }
        return true;
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        freq={}
        if len(s)!=len(t):
            return False
        for ch in s:
            freq[ch]=freq.get(ch,0)+1
        for kh in t:
            freq[kh]=freq.get(kh,0)-1
        for kk in freq.values():
            if kk>0:
                return False
        return True
        